cypress/e2e/challenges.cy.js

```diff
@@ -0,0 +1,91 @@
+import ChallengesPage from '../pages/challengesPage'
+const challengesPage = new ChallengesPage()
+
+describe('Challenge Tests', () => {
+  beforeEach(() => {
+    cy.getEnabledChallenges()
+    cy.getDisabledChallenges()
+  })
+
+  it('Check all enabled challenges load correctly', () => {
+    cy.get('@enabledChallengeIds').then((enabledChallengeIds) => {
+      cy.wrap(enabledChallengeIds).each((challengeNum) => {
+        cy.visit('/')
+        challengesPage.selectChallenge(challengeNum)
+        challengesPage.assertEnabledChallengePage(challengeNum)
+      })
+    })
+  })
+
+  it('Check all disabled challenges load correctly', () => {
+    cy.get('@disabledChallengeIds').then((disabledChallengeIds) => {
+      cy.wrap(disabledChallengeIds).each((challengeNum) => {
```

Comment:
what if the cypress tests are ran on an AWS env with all tests enabled, will this then fail? (e.g. can it handle an empty set?)

Reply:
You know what, I've not tried. I didn't think there would ever be a case that it would happen in the pipeline. 

The code is looking for disabledEnv to not be null. So I guess when its deployed and those challenges are enabled this will be the case right?

``` 
if (response.body.data[i].disabledEnv != null) {
        enabledChallengeIds.push(response.body.data[i].id - 1)
      }
```

Reply:
would it then be null or would it be an empty [] ?